kubernetes_test.go

```diff
@@ -1298,6 +1298,42 @@ func TestKubernetesClusters_GetNodePool(t *testing.T) {
 	require.Equal(t, want, got)
 }
 
+func TestKubernetesClusters_GetNodePoolTemplate(t *testing.T) {
+	setup()
+	defer teardown()
+
+	kubeSvc := client.Kubernetes
+	want := &KubernetesNodePoolTemplateResponse{
+		ClusterUUID: "8d91899c-0739-4a1a-acc5-deadbeefbb8a",
+		Name:        "pool-a",
+		Slug:        "s-1vcpu-2gb",
+		Template: &KubernetesNodePoolTemplate{
+			Labels: map[string]string{
+				"some-label": "some-value",
+			},
+			Capacity: &KubernetesNodePoolResources{
+				CPU:    1,
+				Memory: "2048Mi",
+				Pods:   110,
+			},
+			Allocatable: &KubernetesNodePoolResources{
+				CPU:    390,
+				Memory: "1024Mi",
+				Pods:   110,
+			}},
+	}
+	jBlob, err := json.MarshalIndent(want, "", "   ")
+	require.NoError(t, err)
+	mux.HandleFunc("/v2/kubernetes/clusters/8d91899c-0739-4a1a-acc5-deadbeefbb8a/node_pools_template/pool-a", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodGet)
+		fmt.Fprint(w, string(jBlob))
+	})
+	got, _, err := kubeSvc.GetNodePoolTemplate(ctx, "8d91899c-0739-4a1a-acc5-deadbeefbb8a", "pool-a")
```

Comment:
Another question, only for my own education ✌️ Against what is this sending the request?

Reply:
Ahh it's just sending the request against the handle func above and with that you're basically testing the JSON marshalling/unmarshalling. Sorry for not getting it right away. Correct me if I'm wrong though.

Reply:
Would we in this case not also rather like to write out the raw JSON and not use the structs and marshal that? Because otherwise you'd hide errors potentially as you're using the same implementation in the test and in the code.

Reply:
It's a unit test with a fake handler using httptest https://github.com/digitalocean/godo/blob/21102dd9f3b1518cc230937d73610a4b31a699fd/godo_test.go#L35 in the setup and this is bound to kubeSvc so when the method is called like this it executes against the handler

Reply:
Fixed also as discussed removed json test case parsing and using string literal - Thanks!